scvi/dataset/csv.py

```diff
@@ -1,100 +1,137 @@
-        :new_n_genes: Number of subsampled genes. Default: ``600``.
-        :subset_genes: List of genes for subsampling. Default: ``None``.
-        :compression: For on-the-fly decompression of on-disk data. If ‘infer’ and filepath_or_buffer
-            is path-like, then detect compression from the following extensions: ‘.gz’, ‘.bz2’, ‘.zip’, or ‘.xz’
-            (otherwise no decompression). If using ‘zip’, the ZIP file must contain only one data file to be read in.
-            Default: ``None``.
-        :batch_ids_file: Name of the `.csv` file with batch indices.
-            File contains two columns. The first holds cell names and second
-            holds batch indices - type int. The first row of the file is header.
+class CsvDataset(DownloadableDataset):
+    """Loads a `.csv` file.
+
+    :param filename: Name of the `.csv` file.
+    :param save_path: Save path of the dataset. Default: ``'data/'``.
+    :param url: Url of the remote dataset. Default: ``None``.
+    :param new_n_genes: Number of subsampled genes. Default: ``600``.
+    :param subset_genes: List of genes for subsampling. Default: ``None``.
+    :param compression: For on-the-fly decompression of on-disk data. If ‘infer’ and filepath_or_buffer
+        is path-like, then detect compression from the following extensions: ‘.gz’, ‘.bz2’, ‘.zip’, or ‘.xz’
+        (otherwise no decompression). If using ‘zip’, the ZIP file must contain only one data file to be read in.
+    :param batch_ids_file: Name of the `.csv` file with batch indices.
+        File contains two columns. The first holds cell names and second
+        holds batch indices - type int. The first row of the file is header.
 
     Examples:
         >>> # Loading a remote dataset
-        >>> remote_url = "https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE100866&format=file&file=" \
+        >>> remote_url = "https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE100866&format=file&file="
         ... "GSE100866%5FCBMC%5F8K%5F13AB%5F10X%2DRNA%5Fumi%2Ecsv%2Egz")
         >>> remote_csv_dataset = CsvDataset("GSE100866_CBMC_8K_13AB_10X-RNA_umi.csv.gz", save_path='data/',
-        ... compression='gzip', url=remote_url)
+        ... compression="gzip", url=remote_url)
         >>> # Loading a local dataset
         >>> local_csv_dataset = CsvDataset("GSE100866_CBMC_8K_13AB_10X-RNA_umi.csv.gz",
-        ... save_path='data/', compression='gzip')
-
+        ... save_path="data/", compression='gzip')
     """
 
-    def __init__(self, filename, save_path='data/', url=None, new_n_genes=600, subset_genes=None,
-                 compression=None, sep=',', gene_by_cell=True, labels_file=None,
-                 batch_ids_file=None):
-        self.download_name = filename  # The given csv file is
-        self.save_path = save_path
-        self.url = url
+    def __init__(
+        self,
+        filename: str,
+        save_path: str = "data/",
+        url: str = None,
+        new_n_genes: int = 600,
```

Comment:
Is there a reason for this default? Seems like we are implying 600 is the correct number of genes to use.

Reply:
The only reason is that it is the current default in master: https://github.com/YosefLab/scVI/blob/66cc63ead0d1a219f8d0ba1018d6f610f2b91679/scvi/dataset/csv.py#L37. We don't want to change the behavior of the class because users might already use it.

Reply:
It would definitely mean adding a warning for users already using it.

Reply:
A logging statement + explanation in the docstring should suffice.

Reply:
We don't need to worry a lot about backward compatibility until we hit version 1.0.

Reply:
I removed the weird defaults